cdap-app-templates/cdap-etl/cdap-etl-api-spark/src/main/java/co/cask/cdap/etl/api/batch/SparkMLLib.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.etl.api.batch;
+
+import co.cask.cdap.api.annotation.Beta;
+import co.cask.cdap.api.spark.SparkContext;
+import org.apache.spark.api.java.JavaRDD;
+
+import java.io.Serializable;
+
+
+/**
+ * SparkMLLib composes a final, optional stage of a Batch ETL Pipeline. In addition to configuring the Batch run, it
+ * can also perform RDD operations on the key value pairs provided by the Batch run.
+ *
+ * {@link SparkMLLib#run} method is called inside the Batch Run while {@link SparkMLLib#prepareRun} and
+ * {@link SparkMLLib#onRunFinish} methods are called on the client side, which launches the Batch run, before the
+ * Batch run starts and after it finishes respectively.
+ *
+ * @param <IN> The type of input record to the SparkMLLib.
+ */
+@Beta
+public abstract class SparkMLLib<IN> extends BatchConfigurable<SparkMLLibContext> implements Serializable {
```

Comment:
there's nothing specific to MLLib in this contract.  Perhaps SparkSink is a better name?  But that may imply that it just writes, whereas this can be used to compute then write.  Though I guess that's true about the other sinks too